### PR TITLE
Fix rounding small (but not too small) numbers to zero on calculation (fix #2685)

### LINKED ISSUE #2685: Small numbers are rounded to zero on calculation
**Describe the bug**
In the latest version of jq, `jq -n '1 / 1e-17'` exits with `Division by zero` error, but should emit `1e+17`.

**To Reproduce**
Run `jq -n '1 / 1e-17'`.

**Expected behavior**
Since `1e-17` is not small enough to be rounded to zero, `1e+17` is expected, just like jq 1.6.

**Environment (please complete the following information):**
 - Both macOS, Ubuntu (on Docker).
 - jq version: `jq-1.6-250-gce3701f`

**Additional context**
Likely a regression of cf4b48c7ba30cb30e116b523cff036ea481459f6.

### PR DESCRIPTION
**Pull request by itchyny**
The rounding happens on `decNumberReduce` which reduces the number literal to another decNumber with smaller size to fit to the char buffer we prepare. ~Using `decimal64` from the library, we can avoid reducing.~ Adopted @leonid-s-usov's patch. This PR fixes #2685.

### CODE REVIEW

File: Makefile.am
Comment (leonid-s-usov) [inline Makefile.am:1]: maybe we should discard changes to the file completely? I believe the only change is that you've added jv.h to LIBJQ_INCS but is it required there?

Comment (nicowilliams) [reply Makefile.am:1]: Yeah, why this change?

Comment (leonid-s-usov) [reply Makefile.am:1]: it comes from @itchyny's initial commit. I just removed the new filed from the lists, but haven't reverted the change back

Comment (itchyny) [reply Makefile.am:1]: Ah, okay. I'll drop this.

**Comment by nicowilliams** [comment]
Thanks!

**Comment by nicowilliams** [comment]
@itchyny this crashes with `./jq -cn '5E480463400+9E-6000000099'`.

**Pull request merged**